runners/google-cloud-dataflow-java/worker/src/main/java/org/apache/beam/runners/dataflow/worker/StreamingDataflowWorker.java

```diff
@@ -1585,12 +1585,17 @@ private void getConfigFromDataflowService(@Nullable String computation) throws I
     if (workItem == null || !workItem.isPresent() || workItem.get() == null) {
       return;
     }
-    setMaxWorkItemCommitBytes(180 << 20);
     StreamingConfigTask config = workItem.get().getStreamingConfigTask();
     Preconditions.checkState(config != null);
     if (config.getUserStepToStateFamilyNameMap() != null) {
       stateNameMap.putAll(config.getUserStepToStateFamilyNameMap());
     }
+    if (config.getMaxWorkItemCommitBytes() != null
+        && config.getMaxWorkItemCommitBytes() <= Integer.MAX_VALUE) {
```

Comment:
check it is > 0 too?

otherwise looks good to me, but I think you need to squash your commits into a single one and I'm not a committer

Reply:
Additionally should we only keep the value if it is the global config, not a per-computation config?

Reply:
Done: added check for >= 0.
Done: I switched to only apply the value if it is the global config.
PTAL. Once you are ok with the change I will add a committer.